### DISCUSSION TITLE
Bus/railway platforms should be treated same as footpath/sidewalk for routing

### DISCUSSION THREAD

**Discussion by mdykierek** [osmandapp/OsmAnd#14856] [Ideas]
Consider the following route:
https://www.openstreetmap.org/directions?engine=fossgis_osrm_foot&route=50.26634%2C19.01919%3B50.26589%2C19.01888#map=18/50.26624/19.01932
It is similar to what I get in OsmAnd 4.2.6.
Yet, Graphhoper does it better (IMHO):
https://www.openstreetmap.org/directions?engine=graphhopper_foot&route=50.26634%2C19.01919%3B50.26589%2C19.01888#map=19/50.26621/19.01918

Should OsmAnd walking router consider platforms routable or mappers should consider platforms not routable and put a footpath/sidewalk next to the platform (since platform is for people to get on transportation, it definitely is possible to use it on foot).
I think Graphhoper is more reasonable. Mapping extra paths aside platforms will make the maps cluttered.

Just to let you know, I have reported exact same thing to OSRM. Here is another route to try in Berlin:
https://map.project-osrm.org/?z=17&center=52.518940%2C13.291161&loc=52.518695%2C13.287138&loc=52.519045%2C13.292073&hl=en&alt=0&srv=2

After reviewing some platforms in my area (Silesia and Poland) and giving it a second thought, I find this quite important. I also found a question 11 years old at openstreetmap.org: https://help.openstreetmap.org/questions/4450/mapping-platforms-with-regard-to-routing
Moving around cities involves passing platforms on sidewalks. Currently any platform within a footpath breaks the route badly as if there was a barrier.

**Comment by tommycrock** [comment]
Well, I don't know about other platforms but railway platforms are routable in osmand. e.g. https://www.openstreetmap.org/way/44204179

![Screenshot_20220714-222446_OsmAnd+](https://user-images.githubusercontent.com/5749884/179088800-75aa37f6-7ff8-40f6-9844-542aa2b21b8b.jpg)

**Comment by mdykierek** [comment]
Should have reported this as a defect rather than discussion. Please consider converting.
Double checked it and platforms are routable but OsmAnd is very, very reluctant to take them. It prefers 2 street crossings and taking the opposite side of the road rather than cross a single bus platform. The Berlin example below takes 420m instead 300m.
The Katowice Sokolska example takes 1 signaled crossing, 1 unsafe crossing and 200m instead of straight line 70m sidewalk.
IMHO it should be treated more or less like a piece of sidewalk or footpath rather giving it a "high resistance".
![Screenshot_2022-07-15-10-03-18-948_net osmand](https://user-images.githubusercontent.com/27214784/179181810-9f927ad1-e239-4c44-89e2-c4dcef9dc96a.jpg)
![1657872563526](https://user-images.githubusercontent.com/27214784/179181877-866e5404-4355-4597-b739-695da66999a3.jpg)
